File: testlua_dl/SceneManager.cpp
//
//  SceneManager.cpp
//  ddl_plane
//
//  Created by user on 13-10-30.
//
//

#include "SceneManager.h"

SceneManager *SceneManager::m_sceneManager = NULL;

SceneManager::SceneManager()
{
    
}

SceneManager *SceneManager::sharedSceneManager()
{
    if(m_sceneManager == NULL)
    {
        m_sceneManager = new SceneManager();
    }
    return m_sceneManager;
}

void SceneManager::changeScene(int en_SceneType)
{
    CCScene *pNewScene = NULL;
    
    CCLog("choice select %d", en_SceneType);
    switch (en_SceneType) {
        case en_GameSelectScene:
            pNewScene = GameSelectScene::scene();
            break;
        case en_GameRunScene:
            //pNewScene = GameRunScene::create();
            break;
        case en_GameOverScene:
            break;
        case en_GameWinScene:
            break;
        default:
            break;
    }
    
    if(pNewScene == NULL) return;
    
    CCDirector *pDirector = CCDirector::sharedDirector();
    CCScene *pRuningScene = CCDirector::sharedDirector()->getRunningScene();
    
    if(pRuningScene == NULL)
    {
        pDirector->runWithScene(pNewScene);
    }
    else
    {
        pDirector->replaceScene(pNewScene);
    }
}

File: testlua_dl/SceneManager.lua
require "GameSelectScene"
require "GameRunScene"
require "GameOverScene"

SceneManager = {
	changeScene = function(self, sceneType, data)
		local newScene = nil

		if sceneType == GAME_SELECT_SCENE_TYPE then
			print "select"
			newScene = createGameSelectScene()
		elseif sceneType == GAME_RUN_SCENE_TYPE then
			print "enter game"
			newScene = createGameRunScene()
		elseif sceneType == GAME_OVER_SCENE_TYPE then
			print "game over"
			newScene = createGameOverScene(data)
		elseif sceneType == GAME_WIN_SCENE_TYPE then

		end

		if newScene == nil then
			return
		end

		local director = CCDirector:sharedDirector()
		local runningScene = director:getRunningScene()

		if runningScene == nil then
			print "run scene"
			director:runWithScene(newScene)
		else
			director:replaceScene(newScene)
		end
	end
}

gameOverScene = function(score)
	print "shit"
	SceneManager:changeScene(GAME_OVER_SCENE_TYPE, score)
end

test = function()
	print "test"
end

gameRunScene = function()
	SceneManager:changeScene(GAME_RUN_SCENE_TYPE)
end